### PR TITLE
[ADF-4106] permissions component automated

### PR DESCRIPTION
**Pull request by gmandakini**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/content-services/permissions-component.e2e.ts [hunk 1]
```diff
@@ -25,93 +25,269 @@ import AlfrescoApi = require('alfresco-js-api-node');
...
         done();
     });
 
     afterAll(async (done) => {
         await alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
         await alfrescoJsApi.core.sitesApi.deleteSite(publicSite.entry.id);
-        await alfrescoJsApi.core.groupsApi.deleteGroup(group.entry.id);
+        await alfrescoJsApi.core.sitesApi.deleteSite(privateSite.entry.id);
+        await alfrescoJsApi.core.groupsApi.deleteGroup(groupId);
+        await uploadActions.deleteFilesOrFolder(alfrescoJsApi, roleConsumerFolder.entry.id);
```
Comment (cristinaj) [inline e2e/content-services/permissions-component.e2e.ts]: Same comment as above. All the folders, files could be part of an array or a map, in this way you can use loops in order to delete them, etc.

File: e2e/content-services/permissions-component.e2e.ts [hunk 2]
```diff
@@ -25,93 +25,269 @@ import AlfrescoApi = require('alfresco-js-api-node');
...
-    let duplicateUserPermissionMessage = 'One or more of the permissions you have set is already present : authority -> ' + filePermissionUser.getId() + ' / role -> Contributor';
+    let roleConsumerFolderModel = new FolderModel({'name': 'roleConsumer' + Util.generateRandomString()});
+    let roleCoordinatorFolderModel = new FolderModel({'name': 'roleCoordinator' + Util.generateRandomString()});
+    let roleCollaboratorFolderModel = new FolderModel({'name': 'roleCollaborator' + Util.generateRandomString()});
+    let roleContributorFolderModel = new FolderModel({'name': 'roleContributor' + Util.generateRandomString()});
+    let roleEditorFolderModel = new FolderModel({'name': 'roleEditor' + Util.generateRandomString()});
+    let roleConsumerFolder, roleCoordinatorFolder, roleContributorFolder, roleCollaboratorFolder, roleEditorFolder,
+        siteFolder, privateSiteFile;
+
+    folderOwnerUser = new AcsUserModel();
```
[the remark this answers is not in the document]
Comment (gmandakini) [reply e2e/content-services/permissions-component.e2e.ts]: creating a UserMap seems to be cumbersome, in this PR. we are not only setting the user and role here but also the folders we use are different..

File: e2e/content-services/permissions-component.e2e.ts [hunk 3]
```diff
@@ -34,14 +34,15 @@ import CONSTANTS = require('../util/constants');
 import { MetadataViewPage } from '../pages/adf/metadataViewPage';
 import { UploadDialog } from '../pages/adf/dialog/uploadDialog';
 import { VersionManagePage } from '../pages/adf/versionManagerPage';
+import { forEach } from '@angular/router/src/utils/collection';
```
Comment (adinapitul) [inline e2e/content-services/permissions-component.e2e.ts]: this import is incorrect and not needed anyway

File: e2e/content-services/permissions-component.e2e.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cristinaj) [reply e2e/content-services/permissions-component.e2e.ts]: But the folders used are not related to the role. I suppose you can use an array of folders, without giving them specific names.

**Pull request closed without merging**